### PR TITLE
Bom update artifact version

### PR DESCRIPTION
**Pull request by chhex**
@apgsga-jhe You can already give feedback .... 
I ll be taking on IT-36332 for Testcases

### CODE REVIEW

**Comment by apgsga-jhe** [comment]
@chhex : If I understand it correctly, the aim of "updateArtifact" is to decide if we want to update an Artifact's version or not within a BOM? Other said, to make the difference between a SNAPSHOT and a Patch Build?

File: plugins/revision-manager/src/main/groovy/com/apgsga/revision/manager/domain/RevisionManagerPatchImpl.groovy [hunk 1]
```diff
@@ -14,7 +14,6 @@ class RevisionManagerPatchImpl implements RevisionManager {
     String nextRevision() {
         def currentRev  =revisionPersistence.currentRevision() as Integer
         currentRev++
```
Comment (chhex) [inline plugins/revision-manager/src/main/groovy/com/apgsga/revision/manager/domain/RevisionManagerPatchImpl.groovy:16]: @apgsga-jhe  nextRevision is not persisted automatically anymore

File: plugins/revision-manager/src/main/groovy/com/apgsga/revision/manager/domain/RevisionManagerPatchImpl.groovy [hunk 2]
```diff
@@ -31,6 +30,7 @@ class RevisionManagerPatchImpl implements RevisionManager {
 
     @Override
     void saveRevision(String serviceName, String target, String revision, String fullRevisionPrefix) {
+        revisionPersistence.save(revision as String)
```
Comment (chhex) [inline plugins/revision-manager/src/main/groovy/com/apgsga/revision/manager/domain/RevisionManagerPatchImpl.groovy:33]: @apgsga-jhe But with the explicit saveRevision

File: plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt [hunk 3]
```diff
@@ -35,7 +36,9 @@ open class VersionResolutionExtension(val project: Project, private val revision
     var bomGroupId: String? = null
     var bomBaseVersion: String? = null
     var patchFilePath: String? = null
+    var updateArtifact: String? = null
```
Comment (chhex) [inline plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt:39]: A new variable containing the maven coordinates of the artifact to be updated (or null)

File: plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt [hunk 4]
```diff
@@ -35,7 +36,9 @@ open class VersionResolutionExtension(val project: Project, private val revision
...
     var algorithm: RevisionManagerBuilder.AlgorithmTyp = RevisionManagerBuilder.AlgorithmTyp.SNAPSHOT
+    var newRevision: Boolean = false
```
Comment (chhex) [inline plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt:41]: @apgsga-jhe  Making the publish modus of the Bom explicit , with newRevision we get a new Version

File: plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt [hunk 5]
```diff
@@ -58,31 +61,30 @@ open class VersionResolutionExtension(val project: Project, private val revision
```
Comment (chhex) [inline plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt:61]: @apgsga-jhe  The following is a try to make what is / should be happening more explicit ( and correct ;-)

File: plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt [hunk 6]
```diff
@@ -134,32 +137,32 @@ open class VersionResolutionExtension(val project: Project, private val revision
...
-        assert(bomLastRevision != null) { "lastRevision should not be null" }
-        project.logger.info("BuildVersionResolver with $bomArtifactId, $bomGroupId, $bomBaseVersion and $bomLastRevision")
-        configurationName?.let { configureConfiguration(it) }
+        project.logger.info("Building VersionResolver with $bomArtifactId, $bomGroupId, $bomBaseVersion and Lastrevision:  $lastRevision with new Revision: $newRevision")
+        configurationName.let { configureConfiguration(it) }
         val compositeResolverBuilder = CompositeVersionResolverBuilder()
         project.logger.info("Creating Dependency configuration")
         var cnt = 0
         // Patchfile has higher precedence
+        updateArtifact?.let {
```
Comment (chhex) [inline plugins/version-manager/src/main/kotlin/com/apgsga/maven/dm/ext/VersionResolution.kt:146]: A builder for the update path

File: plugins/version-manager/src/main/kotlin/com/apgsga/maven/impl/resolver/SingleArtifactResolver.kt [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package com.apgsga.maven.impl.resolver
```
Comment (chhex) [inline plugins/version-manager/src/main/kotlin/com/apgsga/maven/impl/resolver/SingleArtifactResolver.kt:1]: The single update path

**Review by chhex — COMMENTED** [review]
@apgsga-jhe  Basically i am ready for a merge.  The Testapps need to be commited merged merged to reflect the changes of this pull.

**Pull request merged**